stacks_queues/queue.py

```diff
@@ -15,47 +15,78 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
         pass
+        if self.size == 0:
+            raise QueueEmptyException('The queue is empty')
+        else:
+            return_value = self.store[self.front]
+            self.store[self.front] = None
+            self.front = (self.front + 1) % self.buffer_size
+            self.size -= 1
+            return return_value
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
         pass
+        if self.size == 0:
+            None
```

Comment:
🤔 Looks like you forgot to return here. As it stands, you basically state the name of a literal `None` and then it floats away into the ether ☁
```suggestion
            return None
```